### PR TITLE
Use a counter instead of relying in XPathNodeIterator.Count

### PR DESCRIPTION
**Pull request by paulomorgado**
Solves #2051

### CODE REVIEW

**Comment by danmosemsft** [comment]
Is it possible to add a unit test?

**Comment by paulomorgado** [comment]
Sure, @danmosemsft,

This was done right in GitHub. Adding a test will require a bit more work.

Where should I add the test?

This will be my first MSBuild Task unit test. Any model I can follow?

**Comment by danmosemsft** [comment]
https://github.com/dotnet/msbuild/blob/f6739274e179fd1ecb7317b8fdc6a3e2c09b4ee0/src/Tasks.UnitTests/XmlPoke_Tests.cs

**Comment by paulomorgado** [comment]
Done!

File: src/Tasks.UnitTests/XmlPoke_Tests.cs [hunk 1]
```diff
@@ -111,6 +111,22 @@ public void PokeChildren()
...
+        {
+            const string original = "b";
+            const string value = "x";
+            const string queryTemplate = "/class/variable[@Name='{0}']/@Name";
+
+            XmlDocument xmlDocument = ExecuteXmlPoke(query: string.Format(queryTemplate, original), value: value);
+
+            List<XmlAttribute> nodes = xmlDocument.SelectNodes(string.Format(queryTemplate, value))?.Cast<XmlAttribute>().ToList();
+
+            Assert.True(nodes?.Count == 1, $"There should be 1 <class /> element with an AccessModifier attribute {Environment.NewLine}{xmlDocument.OuterXml}");
```
Comment (Forgind) [inline src/Tasks.UnitTests/XmlPoke_Tests.cs:125]: nit:
We normally use Shouldly now:
```suggestion
            nodes.Count.ShouldBe(1, $"There should be 1 <class /> element with an AccessModifier attribute {Environment.NewLine}{xmlDocument.OuterXml}");
```

(And nodes also shouldn't be null, right?)

File: src/Tasks.UnitTests/XmlPoke_Tests.cs [hunk 2]
```diff
@@ -111,6 +111,22 @@ public void PokeChildren()
...
+
+            Assert.Equal(value, nodes?.First().Value);
```
Comment (Forgind) [inline src/Tasks.UnitTests/XmlPoke_Tests.cs:127]: nit:
```suggestion
            nodes.First().Value.ShouldBe(value);
```

**Comment by paulomorgado** [comment]
I'll do the changes, they are out of scope, but I'll do them.

Do you want Shouldly to be used on the entire test class or only on the added test method?

**Comment by BenVillalobos** [comment]
@paulomorgado Only the added test method.

Much appreciated!

**Comment by paulomorgado** [comment]
Done!

**Comment by Forgind** [comment]
Thanks!

**Comment by kygagner** [comment]
Using Visual Studio 16.9.2 the issue appears to be fixed - is that expected due to the merged in change, or is this a change in undefined behavior? If the former, we should close the issue.

**Pull request merged**